### PR TITLE
feat: bot balance tracking

### PR DESCRIPTION
**Pull request by ConjunctiveNormalForm**
checks filler balance every 100 blocks and send a cloudwatch metric

later we can try to implement some rebalancing/top-up from within the bot itself it we want to

### CODE REVIEW

File: src/executors/public_1559_executor.rs [hunk 1]
```diff
@@ -241,6 +227,63 @@ where
...
+                    info!("{}- balance: {} at block {}", order_hash, balance_eth.clone(), block_number.as_u64());
+                    if let Some(cw) = &self.cloudwatch_client {
+                        cw.put_metric_data()
+                            .namespace(ARTEMIS_NAMESPACE)
+                            .metric_data(
+                                MetricBuilder::new(CwMetrics::Balance(address.to_string()))
+                                    .add_dimension(
+                                        DimensionName::Executor.as_ref(),
+                                        DimensionValue::PriorityExecutor.as_ref(),
+                                    )
```
Comment (zhongeric) [inline src/executors/public_1559_executor.rs:282]: Should just be one not both dimensions right?

File: src/executors/public_1559_executor.rs [hunk 2]
```diff
@@ -241,6 +227,63 @@ where
...
+                                    .with_value(balance_eth.parse::<f64>().unwrap_or(0.0))
+                                    .build(),
+                            )
+                            .send()
+                            .await?;
```
Comment (zhongeric) [inline src/executors/public_1559_executor.rs]: Hmm maybe we shouldn't await on this like on line 246 above

**Review by zhongeric — APPROVED** [review]

File: src/aws_utils/cloudwatch_utils.rs [hunk 3]
```diff
@@ -77,12 +81,20 @@ pub struct MetricBuilder {
     value: f64,
 }
 
+// TODO: TxStatus type metrics => TxStatus(u32)
```
Comment (zhongeric) [inline src/aws_utils/cloudwatch_utils.rs:84]: can we move TODOs into linear tix?

File: src/executors/public_1559_executor.rs (hunk 1 above)
Comment (ConjunctiveNormalForm) [reply src/executors/public_1559_executor.rs:282]: one is name and the other is value! But yeah, honestly don't need such granularity

File: src/aws_utils/cloudwatch_utils.rs (hunk 3 above)
Comment (ConjunctiveNormalForm) [reply src/aws_utils/cloudwatch_utils.rs:84]: can create linear tix for sure, but I still prefer to have TODOs in code as comments to remind myself from time to time

**Pull request merged**